packages/block-library/src/navigation/edit/index.js

```diff
@@ -191,9 +191,6 @@ function Navigation( {
 		__unstableMarkNextChangeAsNotPersistent,
 	} = useDispatch( blockEditorStore );
 
-	const [ hasSavedUnsavedInnerBlocks, setHasSavedUnsavedInnerBlocks ] =
```

Comment:
We shouldn't need this state in the parent.

Reply:
Yeah, agree I don't think it's needed. I think it was only there to support the previous behavior where a menu would be created whenever `UnsavedInnerBlocks` mounted. From memory there where issues where `UnsavedInnerBlocks` would be unmounted and remounted and could cause duplicate menus to be created—the parent block needed a way to guard the menu creation process.

Now this only happens when a user selects/modifies the inner blocks, and I don't think as much guarding is needed. 👍 

I think it's worth giving this a lot of manual testing to be sure, as it's hard to reason about some aspects of the nav block code and also core data resolution.